include/dgl/runtime/ndarray.h

```diff
@@ -575,6 +580,50 @@ namespace dmlc {
 DMLC_DECLARE_TRAITS(has_saveload, dgl::runtime::NDArray, true);
 }  // namespace dmlc
 
+///////////////// Operator overloading for NDArray /////////////////
+dgl::runtime::NDArray operator + (const dgl::runtime::NDArray& a1,
+                                  const dgl::runtime::NDArray& a2);
+dgl::runtime::NDArray operator - (const dgl::runtime::NDArray& a1,
+                                  const dgl::runtime::NDArray& a2);
+dgl::runtime::NDArray operator * (const dgl::runtime::NDArray& a1,
+                                  const dgl::runtime::NDArray& a2);
+dgl::runtime::NDArray operator / (const dgl::runtime::NDArray& a1,
+                                  const dgl::runtime::NDArray& a2);
+dgl::runtime::NDArray operator + (const dgl::runtime::NDArray& a1, int64_t rhs);
+dgl::runtime::NDArray operator - (const dgl::runtime::NDArray& a1, int64_t rhs);
+dgl::runtime::NDArray operator * (const dgl::runtime::NDArray& a1, int64_t rhs);
+dgl::runtime::NDArray operator / (const dgl::runtime::NDArray& a1, int64_t rhs);
+dgl::runtime::NDArray operator + (int64_t lhs, const dgl::runtime::NDArray& a2);
```

Comment:
So the only support scalar type is integer, does this mean these operators are designed for IdArray actually?

Reply:
That's true. It will raise error for float ndarrays.